### ISSUE TITLE
Error:(107) Error retrieving parent for item: No resource found that matches the given name 'Theme.AppCompat.Light.NoActionBar'. 

### ISSUE THREAD

**Issue by RGiVcardo** [transistorsoft/background-geolocation-lt#29]
## Your Environment
* Plugin version: 2.3.3
* Platform: Android
* OS version: Any
* Device manufacturer / model: Any

## Expected Behavior
I thought i can able to run your sample code demo without any errors.

## Actual Behavior
When I start to build my project then always it gives me error which i have mention in title of this issue. and if i remove dependency from gradle file then no error arrives. but after remove that dependency **"com.intentfilter:android-permissions"** from app run but i saw only blank screen with **Licence validation failure** message.

**Comment by christocracy** [comment]
Looks like an app compact-v4 version issue.

You definitely do not want to remove the android-permissions lib, that’s how the SDK requests location permission.  I’ll look into it.

**Comment by RGiVcardo** [comment]
Sorry dear, but In my gradle file only below app compact dependency available.

compile 'com.android.support:appcompat-v7:28.+'

**Comment by christocracy** [comment]
I’ll look into it within 12 hours.

Btw, in English, “sorry dear” is a phrase typically used between a couple in love or to a young child.

**Comment by christocracy** [comment]
I've posted a new version `0.2.0`, though there's nothing in it responsible for your error.

Your error is related to dependency conflict with Android support libs.

Post the following files:
- `build.gradle`
- `app/build.gradle`

**Comment by christocracy** [comment]
Are you running the Demo app included in this repo?

**Comment by christocracy** [comment]
Use [gradle version 3.2.1](https://github.com/transistorsoft/background-geolocation-lt/blob/master/android/build.gradle#L10)

**Comment by mobiRic** [comment]
I'm experiencing this same issue with the Android Demo app latest 0.2.0 tag, only on `release`.

Exactly as checked out from the repo, with no changes from my side:
- `debug` build works 100% fine.
- `release` build fails with the error.

To be specific, the gradle sync when switching to `release` works fine. Choosing "Build...Rebuild Project" is what triggers the failure.

--------------------------------

I tried some of the usual Support Lib conflict fixes in the `background-geolocation` module gradle file:
- forcing Support Lib versions by explicitly including them
- second attempt, specifying a `resolutionStrategy` (which basically does the same thing)
```
configurations.all {
    resolutionStrategy.force "com.android.support:support-v4:$rootProject.supportLibVersion"
}
```
Neither seems to have any effect.

Checking the dependencies with `gradlew :app:dependencies` shows that all Support Lib versions are at 28, as per your config.

**Comment by mobiRic** [comment]
Gradle task that fails is `:background-geolocation:verifyReleaseResources` which runs `aapt2` (which was known to give issues when first released but can't track down the correct solution).

Output from that task is:
```
org.gradle.internal.UncheckedException: java.util.concurrent.ExecutionException: com.android.builder.internal.aapt.v2.Aapt2Exception: Android resource linking failed
Output:  error: resource style/Theme.AppCompat.Light.NoActionBar (aka com.transistorsoft.backgroundgeolocation:style/Theme.AppCompat.Light.NoActionBar) not found.
error: failed linking references.

```

I have 2 other projects from the last week (my initial demo, and final production project) that give the same behaviour (one of them is using API 27, the other API 28 like this demo).

**Comment by mobiRic** [comment]
By the way, the advice to use [gradle plugin 3.2.1](https://github.com/transistorsoft/background-geolocation-lt/issues/29#issuecomment-434720754) helped in `debug` builds, but as above, not in the `release` build.

**Comment by mobiRic** [comment]
Further information:

Disabling `aapt2` as a temporary workaround does not fix the issue, but points at the android-permissions library `styles.xml` as being the offending file.

Specifically this line:
[`    <style name="AppTheme.Transparent" parent="Theme.AppCompat.Light.NoActionBar">`](https://github.com/nishkarsh/android-permissions/blob/master/src/main/res/values/styles.xml#L3)

With the AAPT error saying:
>AAPT: Error retrieving parent for item: No resource found that matches the given name 'Theme.AppCompat.Light.NoActionBar'.

For some reason that I cannot get to the bottom of, the declaration of the support library within that permissions library doesn't seem to be working.

I've taken the brute force approach that I normally do with support library version compatibility issues - I've added the following lines to the `build.gradle` inside your `background-geolocation` module:
```
    api "com.android.support:appcompat-v7:$rootProject.supportLibVersion"
    api "com.android.support:support-v4:$rootProject.supportLibVersion"
    api "com.android.support:support-media-compat:$rootProject.supportLibVersion"
    api "com.android.support:animated-vector-drawable:$rootProject.supportLibVersion"
```

I'm sure there may be a more elegant approach to solving this issue, but the above solves it for me.

**Comment by christocracy** [comment]
Here's the source for [android-permissions](https://github.com/nishkarsh/android-permissions).  Anything unusual there?  I see not issues there regarding this issue.

**Comment by oferRounds** [comment]
@mobiRic did you manage to solve this issue? Facing it as well

**Comment by darkredz** [comment]
facing the same as well

**Comment by darkredz** [comment]
it was ok with version 2.14.6

**Comment by mobiRic** [comment]
@oferRounds @darkredz
I posted the code that fixed the issue for me in my machine. 

But as I said, I got 4 other people to compile the base code of this library and they did not get this error, which leads me to believe that this is an issue with my (our?) development environment rather than the library.

I did not find a root cause for experiencing this issue, and have not had any problems since making the above change on my machine.
